packages/shared/src/components/PostOptionsReadingHistoryMenu.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { ReactElement, useContext } from 'react';
+import { Item } from '@dailydotdev/react-contexify';
+import dynamic from 'next/dynamic';
+import { QueryClient, QueryKey, useQueryClient } from 'react-query';
+import { ReadHistoryPost } from '../graphql/posts';
+import ShareIcon from '../../icons/share.svg';
+import BookmarkIcon from '../../icons/bookmark.svg';
+import BookmarkIconFilled from '../../icons/filled/bookmark_filled.svg';
+import XIcon from '../../icons/x.svg';
+import useBookmarkPost from '../hooks/useBookmarkPost';
+import AuthContext from '../contexts/AuthContext';
+import { ReadHistoryInfiniteData } from '../hooks/useInfiniteReadingHistory';
+import { useCopyLink } from '../hooks/useCopyLink';
+import { MenuIcon } from './MenuIcon';
+import { QueryIndexes } from '../hooks/useReadingHistory';
+
+const PortalMenu = dynamic(() => import('./fields/PortalMenu'), {
+  ssr: false,
+});
+
+export type PostOptionsReadingHistoryMenuProps = {
+  post: ReadHistoryPost;
+  onHidden?: () => unknown;
+  displayCopiedMessageFunc?: () => void;
+  onHide?: (postId: string) => unknown;
+  indexes: QueryIndexes;
+};
+
+/* eslint-disable no-param-reassign */
+const updateReadingHistoryPost =
+  (
+    queryClient: QueryClient,
+    historyQueryKey: QueryKey,
+    readHistoryPostUpdated: Partial<ReadHistoryPost>,
+    { page, edge }: QueryIndexes,
+  ): ((args: { id: string }) => Promise<() => void>) =>
+  async () => {
+    const oldReadingHistory =
+      queryClient.getQueryData<ReadHistoryInfiniteData>(historyQueryKey);
+    const newItems = [...oldReadingHistory.pages];
+    const history = newItems[page].readHistory.edges[edge].node.post;
+    newItems[page].readHistory.edges[edge].node.post = {
+      ...history,
+      ...readHistoryPostUpdated,
+    };
+    queryClient.setQueryData<ReadHistoryInfiniteData>(historyQueryKey, {
+      ...oldReadingHistory,
+      pages: newItems,
+    });
+    return () => {
+      queryClient.setQueryData<ReadHistoryInfiniteData>(
+        historyQueryKey,
+        oldReadingHistory,
+      );
+    };
+  };
+
+export default function PostOptionsReadingHistoryMenu({
+  post,
+  onHidden,
+  onHide,
+  indexes,
+}: PostOptionsReadingHistoryMenuProps): ReactElement {
+  const { user } = useContext(AuthContext);
+  const queryClient = useQueryClient();
+  const historyQueryKey = ['readHistory', user?.id];
+
+  const [, copyPostLink] = useCopyLink(() => post.commentsPermalink);
+
+  const { bookmark, removeBookmark } = useBookmarkPost({
```

Comment:
I don't remember if `useBookmarkPost` publishes an analytics event but we do need to publish events here on bookmark or remove bookmark.

Reply:
@idoshamun we actually have a `useCopyPostLink` hook with `gaTrackEvent()` inside. Would that be enough?

Reply:
No, we don't need `gaTrackEvent` anymore actually. Maybe @rebelchris can help out with onboarding information for our analytics system

Reply:
Sure we can do it tomorrow if you want?

Reply:
Don't forget to change the actual trackEvent once we have the information.
I've added a TODO in the code for this.